homework/src/index.js

```diff
@@ -33,14 +33,123 @@
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
+      const header = createAndAppend('header', root);
+      createAndAppend('p', header, {html: 'HYF Repositories', class: 'header-lable'});
+      const select = createAndAppend('select', header, { id: 'selectMenu' });
       if (err) {
         createAndAppend('div', root, { html: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        //createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        /* &&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&& */
+        data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {ignorePunctuation: true}));
+        data.forEach((reopsitory, index) => {
+          createAndAppend('option', select, {html: reopsitory.name, value: index});
+        });
+        /* &&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&& */
+        const container = createAndAppend('div', root, { id: 'container'});
+        contentLeft(container, data, select);
+        contentRight(container, data, select);
       }
     });
   }
 
+
+
+
+
+  // function who creates a left content section
+  function contentLeft(container, data, select) {
+    const leftSection = createAndAppend('div', container, { id: 'leftSection', class: 'whiteframe' });
+    showTableRepo(leftSection, data, 0);
+
+    select.addEventListener('change', (event) => {
+      document.getElementById('leftSection').innerHTML = "";
+      const theEvent = event.target.value;
+      showTableRepo(leftSection, data, theEvent);
+    });
+  }
+
+  
+
+  // drow the table
+function showTableRepo(leftSection, data, dataIndex) {
```

Comment:
If you just pass the selected repo instead of the whole array and the index of the selected repo, your argument list can be simpler and the body of the function can just use the selected repo without having to index it from the array. e.g.:

```js
function showTableRepl(leftSection, repo) {
  //...
  let nameLink = '<a href="' +repo.html_url + ...
  //...
}
```

Reply:
Here i need the leftSection variable to generate the container and also the table of the contents in it